### PR TITLE
Add fatal check to catch index out of bounds error properly

### PR DESCRIPTION
**Pull request by dfquaresma**
#### What is the purpose of this pull request?

This PR improves Replayer's error check to catch an index out of bounds properly to us, making easy for our life to debug the error.

#### What problem is this solving?

When the Replayer faces an input entry out of the expected format, it may raise an `index out of bounds` error. 

#### How should this be manually tested?
You may write short input files in the right format and wrong one, and run this version of Replayer with it. 

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: replayer/main.go [hunk 1]
```diff
@@ -38,6 +38,24 @@ func fatalCheckError(msg string, err error, line int) {
 	}
 }
 
+func fatalEntryCheckError(req, reqs []string, i int) {
```
Comment (paulofelipefeitosa) [inline replayer/main.go]: Could you please give a more meaningful name for these parameters? Like `reqComponents`.

Comment (paulofelipefeitosa) [inline replayer/main.go]: Do you really need all the requests to evaluate whether the entry is invalid or not? What you are doing is logging the previous and next lines for debugability, this function could just check if the entry is invalid or not and the logging logic could be handled at the main function.

Comment (thiagomanel) [reply replayer/main.go]: generally speaking, I agree with @paulofelipefeitosa. However, I'm not sure the code will be simpler. @paulofelipefeitosa, do you mean, returning the error msg (and the other error params) to the callee?

Comment (thiagomanel) [reply replayer/main.go]: (btw, being opinionated here ... I'd output only one line (or even the line number). The user could `grep -A -B` to help himself

Comment (paulofelipefeitosa) [reply replayer/main.go]: @thiagomanel I mean to just return an error that indicates if the entry is valid or not. Then, the debugging logic to log the previous and next line could be done at the callee (if necessary).

**Comment by dfquaresma** [comment]
@thiagomanel and @paulofelipefeitosa , thanks for your review! I've read and applied modifications based on your comments.

**Pull request merged**